vcr/concept/template.go

```diff
@@ -53,11 +59,19 @@ type Template struct {
 	// reverseIndexMapping contains mappings from VC JSON joinPath to concept JSON joinPath
 	reverseIndexMapping map[string]string
 	// indices contains a set of compound indices
-	indices [][]string
+	indices [][]CompoundIndex
 	// fixedValues contains key/value pairs that always need to be added to the backend query
 	fixedValues map[string]string
 }
 
+// CompoundIndex holds the type of index for a particular concept JSON path
+type CompoundIndex struct {
```

Comment:
I think the name is a bit misleading, because it implies it's a combination of multiple indices while it's just a data structure to track indices with their desired types parsed from the JSON template? So maybe just `TemplateIndex`/`ConceptIndex`/`TemplateConceptIndex`?

Reply:
switched to `TemplateIndex`